numpy/lib/histograms.py

```diff
@@ -237,6 +237,10 @@ def _get_outer_edges(a, range):
         first_edge = first_edge - 0.5
         last_edge = last_edge + 0.5
 
+    if a.dtype.kind not in ['i']:
```

Comment:
Perhaps this should be `np.issubdtype(a.dtype, np.inexact)`?

Reply:
Add a comment here why we are doing this: I think we're still ensuring ufunc casting does not cause problems...

Reply:
An alternative fix might just be to promote the bounds to 0d arrays